### PR TITLE
Fix broken login from UI

### LINKED ISSUE #3416: Login with user@realm in webui not possible
This is a problem on the master branch we are currently working on.

### PR DESCRIPTION
**Pull request by plettich**
When a user logs in via the WebUI, there is always a `realm` parameter added to the `/auth` request, even if no realm is given (empty string). With the changes from fa643349 this lead to the user not being found if the login was `user@realm` and `realm` wasn't the default realm.

Closes #3416

### CODE REVIEW

File: tests/test_api_auth.py [hunk 1]
```diff
@@ -120,6 +123,8 @@ def test_01_auth_with_split(self):
             # realm1 should be the default realm
             self.assertEqual('realm1', result['value']['realm'], result)
 
+        # TODO: Add test with empty realm parameter and user@realm not in default realm
```
Comment (cornelinux) [inline tests/test_api_auth.py:126]: Are you still planning on doing this? Like on Friday?

**Review by cornelinux — COMMENTED** [review]
Looks good. Would you also add the TODO test?

**Review by cornelinux — APPROVED** [review]
